pkg/agent/controller/networkpolicy/reject_test.go

```diff
@@ -502,3 +522,180 @@ func Test_getRejectPacketOutMutateFunc(t *testing.T) {
+				// EpSelectedRegMark
+				Class:          openflow15.OXM_CLASS_PACKET_REGS,
+				Field:          2,
+				HasMask:        false,
+				Length:         0,
+				ExperimenterID: 0,
+				Value: &openflow15.ByteArrayField{
+					Data:   []byte{0, 2, 0, 0, 0, 0, 0, 0},
+					Length: 64,
+				},
+				Mask: nil,
+			}}),
+			antreaProxyEnable: true,
+			prepareFunc: func(mockClient *oftesting.MockClient, mockIStore *storetesting.MockInterfaceStore) {
+				mockIStore.EXPECT().GetInterfaceByIP(gomock.Any()).Return(nil, false)
+				mockIStore.EXPECT().GetInterfaceByIP(gomock.Any()).Return(nil, false)
+				mockClient.EXPECT().SendTCPPacketOut(fakeDstMac, openflow.GlobalVirtualMAC.String(), fakeDstIPv4, fakeSrcIPv4, gomock.Any(), gomock.Any(), false, fakeDstPort, fakeSrcPort, gomock.Any(), fakeTCPSeqNum+1, gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Times(1).Return(nil)
+			},
+		},
+		{
+			name:              "RejectPodLocalIPv4UDPTrafficWithAntreaProxy",
+			packetIn:          genPacketIn(fakeSrcMac, fakeDstMac, fakeSrcIPv4, fakeDstIPv4, fakeSrcPort, fakeDstPort, 0, false, false, []openflow15.MatchField{*openflow15.NewInPortField(uint32(fakeOFPort1))}),
+			antreaProxyEnable: true,
+			prepareFunc: func(mockClient *oftesting.MockClient, mockIStore *storetesting.MockInterfaceStore) {
+				mockIStore.EXPECT().GetInterfaceByIP(gomock.Any()).Return(fakeSIface, true)
+				mockIStore.EXPECT().GetInterfaceByIP(gomock.Any()).Return(fakeDIface, true)
+				mockClient.EXPECT().SendICMPPacketOut(fakeDstMac, fakeSrcMac, fakeDstIPv4, fakeSrcIPv4, gomock.Any(), gomock.Any(), false, uint8(3), uint8(10), gomock.Any(), gomock.Any()).Times(1).Return(nil)
+			},
+		},
+		{
+			name:              "RejectPodLocalIPv6TCPTrafficWithAntreaProxy",
+			packetIn:          genPacketIn(fakeSrcMac, fakeDstMac, fakeSrcIPv6, fakeDstIPv6, fakeSrcPort, fakeDstPort, fakeTCPSeqNum, true, true, nil),
+			antreaProxyEnable: true,
+			prepareFunc: func(mockClient *oftesting.MockClient, mockIStore *storetesting.MockInterfaceStore) {
+				mockIStore.EXPECT().GetInterfaceByIP(gomock.Any()).Return(fakeSIface, true)
+				mockIStore.EXPECT().GetInterfaceByIP(gomock.Any()).Return(fakeDIface, true)
+				mockClient.EXPECT().SendTCPPacketOut(fakeDstMac, fakeSrcMac, fakeDstIPv6, fakeSrcIPv6, gomock.Any(), gomock.Any(), true, fakeDstPort, fakeSrcPort, gomock.Any(), fakeTCPSeqNum+1, gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Times(1).Return(nil)
+			},
+		},
+	}
+	ctrl := gomock.NewController(t)
+	controller, _, _ := newTestController()
+	mockClient := oftesting.NewMockClient(ctrl)
+	mockIStore := storetesting.NewMockInterfaceStore(ctrl)
+	controller.ofClient = mockClient
+	controller.ifaceStore = mockIStore
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			controller.antreaProxyEnabled = tt.antreaProxyEnable
```

Comment:
Will there be a race condition if these test cases are execution in parallel?  I think you should move lines 628-633 inside of `t.Run()`.

Reply:
Done.